client/allocrunner/taskrunner/vault_hook.go

```diff
@@ -320,23 +337,77 @@ OUTER:
 
 // deriveVaultToken derives the Vault token using exponential backoffs. It
 // returns the Vault token and whether the manager should exit.
-func (h *vaultHook) deriveVaultToken() (token string, exit bool) {
+func (h *vaultHook) deriveVaultToken() (string, bool) {
 	var attempts uint64
 	var backoff time.Duration
+	var err error
+	var exit bool
+	var token string
+
 	for {
-		tokens, err := h.client.DeriveToken(h.alloc, []string{h.taskName})
-		if err == nil {
-			return tokens[h.taskName], false
+		switch h.authMethod {
+		case vaultAuthMethodToken:
+			token, exit, err = h.deriveVaultTokenLegacy()
+		case vaultAuthMethodJWT:
+			token, exit, err = h.deriveVaultTokenJWT()
 		}
+		if err == nil || exit {
+			fmt.Println(err)
```

Comment:
Debugging leftovers:

```suggestion
```

Reply:
Ah yes, I left this to remind myself that I needed to handle this error (it isn't immediately obvious that can also hit this path when `err != nil`).

But I refactored this code to make it simpler and more obvious when errors need to be handled 👍